addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/type/MetadataUtils.java

```diff
@@ -65,27 +68,45 @@ private static void loadBundle(String filename) {
      * Loads the standard datapoints for channel metadata generation.
      */
     private static void loadStandardDatapoints() {
-        try (InputStream is = Thread.currentThread().getContextClassLoader()
-                .getResourceAsStream("homematic/standard-datapoints.properties")) {
-            BufferedReader reader = new BufferedReader(new InputStreamReader(is));
-
-            String line;
-            while ((line = reader.readLine()) != null) {
-                if (StringUtils.trimToNull(line) != null && !StringUtils.startsWith(line, "#")) {
-                    String channelType = StringUtils.trimToNull(StringUtils.substringBefore(line, "|"));
-                    String datapointName = StringUtils.trimToNull(StringUtils.substringAfter(line, "|"));
-
-                    Set<String> channelDatapoints = standardDatapoints.get(channelType);
-                    if (channelDatapoints == null) {
-                        channelDatapoints = new HashSet<String>();
-                        standardDatapoints.put(channelType, channelDatapoints);
+        try {
+            Bundle bundle = FrameworkUtil.getBundle(MetadataUtils.class);
+            InputStream stream = null;
+            BufferedReader reader = null;
+            try {
+                stream = bundle.getResource("homematic/standard-datapoints.properties").openStream();
+                reader = new BufferedReader(new InputStreamReader(stream));
+
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    if (StringUtils.trimToNull(line) != null && !StringUtils.startsWith(line, "#")) {
+                        String channelType = StringUtils.trimToNull(StringUtils.substringBefore(line, "|"));
+                        String datapointName = StringUtils.trimToNull(StringUtils.substringAfter(line, "|"));
+
+                        Set<String> channelDatapoints = standardDatapoints.get(channelType);
+                        if (channelDatapoints == null) {
+                            channelDatapoints = new HashSet<String>();
+                            standardDatapoints.put(channelType, channelDatapoints);
+                        }
+
+                        channelDatapoints.add(datapointName);
                     }
-
-                    channelDatapoints.add(datapointName);
                 }
+            } catch (NullPointerException | IllegalStateException | IOException e) {
+                throw new IOException("The resource homematic/standard-datapoints.properties could not be loaded!", e);
+            } finally {
+                closeQuietly(reader);
+                closeQuietly(stream);
             }
-        } catch (IOException ex) {
-            logger.warn("Can't load standard-datapoints.properties file!");
+        } catch (Exception e) {
```

Comment:
Please don't catch `Exception`'s, the `IOException` was fine. Or explain me a good reason to change it.